kaldb/src/main/java/com/slack/kaldb/writer/kafka/KaldbKafkaConsumer.java

```diff
@@ -255,7 +244,7 @@ public boolean consumeMessagesBetweenOffsetsInParallel(
     LOG.info("Pool size for queue is: {}", poolSize);
 
     // TODO: Track and log errors and success better.
-    BlockingArrayBlockingQueue<Runnable> queue = new BlockingArrayBlockingQueue<>(100);
+    ArrayBlockingQueue<Runnable> queue = new ArrayBlockingQueue<>(100);
```

Comment:
Can we add a unit test to this use case to make sure the code works as expected?

Reply:
I think there will need to be some restructuring of the code to be able to test that particular change in isolation.

Reply:
I think it's worth doing it since I am not quite sure looking at the code if it will work as intended.